api/clients/eigenda_client.go

```diff
@@ -70,6 +79,17 @@ func NewEigenDAClient(log log.Logger, config EigenDAClientConfig) (*EigenDAClien
 		return nil, err
 	}
 
+	var ethClient *ethclient.Client
+	var edasmCaller *edasm.ContractEigenDAServiceManagerCaller
+	ethClient, err = ethclient.Dial(config.EthRpcUrl)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial ETH RPC node: %w", err)
+	}
+	edasmCaller, err = edasm.NewContractEigenDAServiceManagerCaller(common.HexToAddress(config.SvcManagerAddr), ethClient)
```

Comment:
No, but I still made the eth-rpc and service-manager address mandatory, because I realized we should probably be checking the confirmed status of the batch onchain when the disperser tells us (defensive programming).

Realized that it didn't make sense to require them only when confirmationDepth > 0, because why also not verify that the batch has landed onchain when confirmationDepth = 0 (meaning its included only in the currentBlock)? See https://github.com/Layr-Labs/eigenda/blob/7f7fdaf885be3046c006002a1edb073248518827/api/clients/eigenda_client.go#L270

Reply:
@epociask is that OK with you or do you see a problem with this?

Reply:
Talked offline. We are fine with this. Added a breaking change comment to the PR description.